mpc/mpc.py

```diff
@@ -607,16 +637,70 @@ def main():
     global logtime
     logcrs = log.CRS
     logtime = log.TIME
-    log.to_csv('l.csv', index=False)
+    log.rename(columns={"LAT": "y", "LON": "x"}, inplace=True)
+
+    return log
+
+
+def mpc(path, length, width, backtowheel, wb, wheel, target_speed, max_speed, max_accel, max_steer, max_dsteer):
+    global LENGTH
+    LENGTH = float(length)
+    global WIDTH
+    WIDTH = float(width)
+    global BACKTOWHEEL
+    BACKTOWHEEL = float(backtowheel)
+    global WB
+    WB = float(wb)
+    global WHEEL_LEN
+    global WHEEL_WIDTH
+    WHEEL_LEN = float(wheel)
+    WHEEL_WIDTH = float(wheel)
+    global TARGET_SPEED
+    TARGET_SPEED = float(target_speed) / 3.6
+    global MAX_STEER
+    MAX_STEER = np.deg2rad(float(max_steer))
+    global MAX_SPEED
+    MAX_SPEED = float(max_speed) / 3.6
+    global MAX_ACCEL
+    MAX_ACCEL = float(max_accel)
+    global MAX_DSTEER
+    MAX_DSTEER = np.deg2rad(float(max_dsteer))
+    return mains(path)
+
+
+def mains(path):

```

Comment:
What exactly is mains()? Seems a bit redundant since you have main() just below, or maybe just add some comments to explain it.